### DISCUSSION TITLE
SQLite

### DISCUSSION THREAD

**Discussion by huwgit** [vendurehq/vendure#1938] [Q&A]
I am building an application using Vendure and SQLite. I would expect a maximum number of orders/writes of 10 per day. Given this would SQLite be adequate for this kind of traffic? What other issues might I face using this database. I am thinking of migrating to postgres but ideally I'd like to stay where I am.

**Comment by huwgit** [comment]
I am more familiar with frontend stuff I should mention,.

I am getting the foillowing when running with npm init @vendure my-app and setting postgres. I can see that vendure database has been created and I have added a user root (grant all priveleges) to this database manually. 


  ✖ Initializing server
    → [connect ECONNREFUSED ::1:5432]
npm ERR! code 1
npm ERR! path /var/lib/postgresql
npm ERR! command failed
npm ERR! command sh -c create "my-vendure3"

My .env vile is as it should be - any ideas why Im getting this error..

**Reply by michaelbromley** [reply]
Hi!
In answer to your original question: for ~10 orders per day. SQLite should be fine. The Vendure demo at demo.vendure.io, as well as all our demo storefronts, as backed by SQLite.

The error about connecting to postgres suggests the connection settings are not set up right. Either the port, or credentials, or something.

**Reply by huwgit** [reply]
In the log I get this error:

33 verbose stack Error: command failed
33 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/index.js:64:27)
33 verbose stack     at ChildProcess.emit (node:events:527:28)
33 verbose stack     at maybeClose (node:internal/child_process:1090:16)
33 verbose stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)

**Reply by huwgit** [reply]
If I need to migrate in 2 years time from SQLite to Postgres - that  is doable? (once I learn more about backend stuff!!)

**Reply by michaelbromley** [reply]
Yes, migrating should indeed be possible. There are even tools dedicated to this: https://pgloader.readthedocs.io/en/latest/ref/sqlite.html
